### DISCUSSION TITLE
When Netty or Netty 5 could provide JPMS(module-info.java) support?

### DISCUSSION THREAD

**Discussion by chengenzhao** [netty/netty#12819] [Q&A] [answered]
Hi:

We are using Vert.x in our JavaFX projects
everything works fine
except it doesn't support JPMS thus we have to abandon it in our client side projects for the reason of jlink and native image which require jar to provide module-info.java
so sad because Vert.x provide dozens of clients e.g. tcp, udp, mqtt, web socket etc.
There are only two dependencies of Vert.x: Netty and Jackson which already support JPMS, we all happy with it
so when Netty could provide this feature?

Vert.x clients are really helpful for those client side projects
but it has to wait Netty or Netty 5 support JPMS first

Thanks

**Comment by chrisvest** [comment, marked as answer]
We haven't yet decided if Netty 5 should declare Java Platform Modules, or continue to use Automatic Module Names like Netty 4.1.

**Reply by chengenzhao** [reply]
Thanks for the reply
AFAIK, Jackson and Kotlin does provide JPMS support long ago and Helidon promises to delivery Nima in the next year
so Netty 5 should really take this option into consideration
Good Luck
